docs: update deploy docs to include new flags and functionality (#5260)
aws/copilot-cli: site/content/docs/commands/deploy.en.md

## What does it do?

> This command is used to get you from local manifests to deployed service and environment. It will check for deployed infrastructure and local manifests, help you initialize and deploy an environment, and deploy a workload

Reviewer: ```suggestion
This command is used to get you from local manifests to deployed service and environment. It will check for deployed infrastructure and local manifests, help you initialize and deploy an environment, and deploy a workload.
```

If a workload is uninitialized, `--init-wkld` will initialize the workload before deploying it.

If the desired environment is uninitialized, you can initialize it with `--init-env`. 

The `--deploy-env` flag can be specified to skip environment deployment confirmation, or can be set to false (`--deploy-env=false`) to skip 
deploying the environment.

> The resteps involved in `copilot deploy` are as follows:

Reviewer: ```suggestion
The steps involved in `copilot deploy` are as follows:
```

## What does it do?

> This command is used to run either [`copilot svc deploy`](../commands/svc-deploy.en.md) or [`copilot job deploy`](../commands/job-deploy.en.md) under the hood.

Reviewer: i think we can just get rid of this and rely on the line below (maybe need to reword it a bit, but) - it's more complicated that this now

It also checks for service and environment manifests in the local workspace, and allows you to initialize and deploy workloads and environments.

If a workload is uninitialized, `--init-wkld` will initialize the workload before deploying it.

> If the desired environment is uninitialized, you may initialize it with `--init-env`. 

Reviewer: ```suggestion
If the desired environment is uninitialized, you can initialize it with `--init-env`. 
```

The `--deploy-env` flag can be specified to skip environment deployment confirmation, or can be set to false (`--deploy-env=false`) to skip 
deploying the environment.

> The steps involved in `copilot deploy` are the same as those involved in `copilot svc deploy` and `copilot job deploy`:

Reviewer: like i mentioned above - this isn't really true anymore as we do stuff with environments and initializing workloads

> 1. If your service does not exist, optionally initialize it (via `--init-env[=false]`).

Reviewer: ```suggestion
1. If your service does not exist, optionally initialize it (via `--init-wkld`)
```
?

> 2. If the target environment does not exist, optionally initialize it with custom credentials (via `--init-env[=false]`, `--region`, `--profile`, or `--aws-*` flags).

Reviewer: ```suggestion
2. If the target environment does not exist, optionally initialize it with custom credentials (via `--init-env`, `--region`, `--profile`, or `--aws-*` flags).
```

> 3. Optionally deploy the environment before service deployment (via `--deploy-env[=false]`).

Reviewer: actually i'm wondering if we should just not show the flags in this numbered section 🤔 we mentioned them above and below already, and we follow the logic whether or not the flags are present, just prompting instead.

4. When `image.build` exists in the manifest:
    1. Build your local Dockerfile into an image
    2. Tag it with the value from `--tag` or the latest git sha (if you're in a git directory)
    3. Push the image to ECR
5. Package your manifest file and addons into CloudFormation.
6. Create / update your ECS task definition and job or service.

## What are the flags?

```
      --allow-downgrade                Optional. Allow using an older version of Copilot to update Copilot components
                                       updated by a newer version of Copilot.
  -a, --app string                     Name of the application.
      --aws-access-key-id string       Optional. An AWS access key for the environment account.
      --aws-secret-access-key string   Optional. An AWS secret access key for the environment account.
      --aws-session-token string       Optional. An AWS session token for temporary credentials.
      --deploy-env bool                Deploy the target environment before deploying the workload.

>       --detach bool                    Optional. Skip displaying CloudFormation deployment progress.

Reviewer 2: Do we need these `bool` at the end?

Author: I think so, just for consistency with other flag types. I'd prefer to keep it explicit even if the help docs don't render them when you do `copilot deploy -h`.

Reviewer 2: I feel like the `bool` at the end might be misleading for `copilot deploy --detach true`...but just a nit im ok for both!

rollback of the stack via the AWS console or AWS CLI before the next deployment.

## Examples

> Deploys a service named "frontend" to a "test" environment.

Reviewer: ```suggestion
Deploys a service named "frontend" to an existing "test" environment.
```

```console

> $ copilot deploy --name frontend --env test --deploy-env=false

Reviewer: do they _have_ to have `--deploy-env=false` if the environment already exists?

Author: it'll prompt you to see if you want to deploy the environment first.

Author: We  could skip that and instead make deploying the env fully optional and opt-in but It'd be much less discoverable i think

Reviewer: wait - so if i have _already deployed_ the `test` environment before running `copilot deploy`, it'll still prompt me to deploy it, unless i have the flag?

Author: yeah. maybe i need to flop that behavior? make it opt in? or do that immediately following this in a patch?

Reviewer 2: Yeah i remember i had similar feedback maybe in the bug bash? I feel like we can probably improve in our next release as it's not a blocker imo.

```

Deploys a job named "mailer" with additional resource tags to a "prod" environment.
```console
$ copilot deploy -n mailer -e prod --resource-tags source/revision=bb133e7,deployment/initiator=manual --deploy-env=false
```

Initializes and deploys an environment named "test" in us-west-2 under the "default" profile with local manifest,
then deploys a service named "api"
```console

> $ copilot deploy --init-env --deploy-env --env test --name api --profile default --region us-west-2

Reviewer 3: It occurs to me that `--deploy-env` sort of implies `--init-env`! If a user run `copilot deploy --deploy-env --env test`, where `test` is not initialized yet, perhaps it is totally fair for us to just initialize & deploy `test`, even if `--init-env` isn't specified 💭 

This is non-blocking though! If that makes sense to we all, we probably can do this in a future release!
